urlc58f2e0b85_xoriole_CEAC_tgz-pJ8-com_sdm_phr_cpabe_CpabeTestJ8/src/com/sdm/phr/CryptoUtil.java

```diff
@@ -0,0 +1,60 @@
+package com.sdm.phr;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.security.MessageDigest;
+
+public class CryptoUtil {
+
+    public static String getFileChecksum(String filename, String type) {
+        if (filename.equalsIgnoreCase("N/A")) {
+            return "null";
+        }
+        String checksum = null;
+        try {
+            MessageDigest md = MessageDigest.getInstance(type);
+            FileInputStream fis = new FileInputStream(filename);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *fis*. Execution paths that do not contain closure statements were detected. Ensure *fis* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).